src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -167,21 +169,26 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
             {
                 try
                 {
-                    // Check if initial configuration load had failed
-                    if (_applicationSettings == null)
-                    {
-                        if (InitializationCacheExpires < DateTimeOffset.UtcNow)
+                    async Task funcToExecute(ConfigurationClient client) {

```

Comment:
What do you think about adding an overload to prevent the need for inline function? I think usage would be like

```

public async Task RefreshAsync(CancellationToken cancellationToken)
{
    await ExecuteWithFailOverPolicyAsync(
        (client) =>
            RefreshAsync(
                client,
                cancellationToken),
        cancellationToken)
        .ConfigureAwait(false);
}
```

Reply:
I had a concern about ensuring there is only one network operation in progress in the new overload.
But I put a comment there explaining the caller needs to take care of this, so we should be good.